PatientWebApplication/SearchMicroserviceApi/Service/AppointmentSearchService.cs

```diff
@@ -0,0 +1,216 @@
+        {
+            if (!UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.DoctorNameAndSurname))
+            {
+                appointments = appointments.FindAll(appointment => appointment.DoctorNameAndSurname.Contains(appointmentSearchDto.DoctorNameAndSurname));
+            }
+            return appointments;
+        }
+
+        /// <summary> This method is getting list of filtered <c>DoctorAppointment</c> of one patient by date. </summary>
+        /// /// <param name="appointments"><c>appointments</c> is List of appointments that matches search fields.
+        /// </param>
+        /// <param name="appointmentReportSearchDto"><c>appointmentReportSearchDto</c> is Data Transfer Object of a <c>Operation</c> that is being used to filter operations.
+        /// </param>
+        /// <returns> List of filtered patient's appointments. </returns>
+        private List<AppointmentDto> SearchForDate(List<AppointmentDto> appointments, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (!UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.Start) && !UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.End))
+            {
+                appointments = GetAppointmentsBetweenDates(appointmentSearchDto.Start, appointmentSearchDto.End, appointments);
+            }
+            else if (UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.Start) && !UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.End))
+            {
+                appointments = GetAppointmentsBeforeDate(appointmentSearchDto.End, appointments);
+            }
+            else if (!UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.Start) && UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.End))
+            {
+                appointments = GetAppointmentsAfterDate(appointmentSearchDto.Start, appointments);
+            }
+            return appointments;
+
+        }
+
+        /// <summary> This method is getting list of filtered <c>DoctorAppointment</c> of one patient that are between two dates. </summary>
+        /// <param name="appointments"><c>appointments</c> is List of appointments that matches search fields.
+        /// </param>
+        /// <param name="start"><c>start</c> is first date of search.
+        /// </param>
+        /// <param name="end"><c>end</c> is last date of search.
+        /// </param>
+        /// <returns> List of filtered patient's appointments. </returns>
+        private List<AppointmentDto> GetAppointmentsBetweenDates(string start, string end, List<AppointmentDto> appointments)
+        {
+            DateTime startDate = UtilityMethods.ParseDateInCorrectFormat(start);
+            DateTime endDate = UtilityMethods.ParseDateInCorrectFormat(end);
+            return appointments.FindAll(appointment => startDate <= UtilityMethods.ParseDateInCorrectFormat(appointment.Date) && UtilityMethods.ParseDateInCorrectFormat(appointment.Date) <= endDate);
```

Comment:
Write this in form:
"
return appointments.FindAll(appointment =>UtilityMethods.ParseDateInCorrectFormat(start) <= UtilityMethods.ParseDateInCorrectFormat(appointment.Date) && UtilityMethods.ParseDateInCorrectFormat(appointment.Date) <= UtilityMethods.ParseDateInCorrectFormat(start));
"